lib/trento/application/integration/discovery/policies/host_policy.ex

```diff
@@ -209,4 +302,91 @@ defmodule Trento.Integration.Discovery.HostPolicy do
   defp parse_storage_profile(%{data_disks: nil}), do: 0
   defp parse_storage_profile(%{data_disks: data_disks}), do: length(data_disks)
   defp parse_storage_profile(_), do: 0
+
+  # Saptune payload
+
+  defp format_saptune_payload_keys(map) when is_map(map) do
+    try do
+      for {key, val} <- map,
+          into: %{},
+          do: {snake_case(key), format_saptune_payload_keys(val)}
+    rescue
+      Protocol.UndefinedError -> map
+    end
+  end
+
+  defp format_saptune_payload_keys(list) when is_list(list) do
+    list
+    |> Enum.map(&format_saptune_payload_keys/1)
+  end
+
+  defp format_saptune_payload_keys(other_types), do: other_types
+
+  defp snake_case(payload) when is_binary(payload) do
+    payload |> String.replace(" ", "_") |> String.downcase()
+  end
+
+  defp format_saptune_services_list(services) when map_size(services) == 0, do: []
```

Comment:
Do we need this clause?
I would expect that the next one, with the `Enum.map` with and empty `service_map` returning an empty list

Reply:
done